### PR TITLE
[DOCS] Adds HTTP response count example to Painless examples

### PR DESCRIPTION
**Pull request by szabosteve**
This PR contains a scripted metric aggregation example for counting HTTP responses in a web log dataset and adds it to the Painless examples for transforms page.

### CODE REVIEW

File: docs/reference/transform/painless-examples.asciidoc [hunk 1]
```diff
@@ -327,3 +327,63 @@ the buckets you want to use for the variable. In this particular case, `min` and
...
+<2> Object of the `scripted_metric` aggregation.
+<3> This `scripted_metric` performs a distributed operation on the web log data 
+to count specific types of HTTP responses (error, success, and other).
+<4> The `init_script` creates a `responses` array in the `state` object with 
+three properties (`error`, `success`, `other`) with long data type.
+<5> The `map_script` defines `code` based on the `response.keyword` value of the 
+document, then it counts the errors, successes, and other responses based on the 
+first digit of the responses.
+<6> The `combine_script` returns `state.responses` from each shard.
+<7> The `reduce_script` craters a `counts` array with the `error`, `success`,
```
Comment (davidkyle) [inline docs/reference/transform/painless-examples.asciidoc]: craters -> create ??

Comment (szabosteve) [inline docs/reference/transform/painless-examples.asciidoc]:
```suggestion
<7> The `reduce_script` creates a `counts` array with the `error`, `success`, 
```

Comment (szabosteve) [reply docs/reference/transform/painless-examples.asciidoc]: I honestly don't know how it happened. Thanks @davidkyle for spotting it! :)

File: docs/reference/transform/painless-examples.asciidoc [hunk 2]
```diff
@@ -327,3 +327,63 @@ the buckets you want to use for the variable. In this particular case, `min` and
...
+==== Counting HTTP responses by using scripted metric aggregation
+
+You can count the different HTTP response types in a web log data set by using 
+scripted metric aggregation as part of the {transform}. The example below 
+assumes that the HTTP response codes are stored in the `response` field of the 
+documents.
+
+[source,js]
+--------------------------------------------------
+"aggregations": { <1>
```
Comment (hendrikmuhs) [inline docs/reference/transform/painless-examples.asciidoc]: we omit "aggregations" on the other example. I think we should be consistent. However I like it, so I suggest to add "aggregations" on the other example.

File: docs/reference/transform/painless-examples.asciidoc [hunk 3]
```diff
@@ -327,3 +327,63 @@ the buckets you want to use for the variable. In this particular case, `min` and
...
+
+
+[discrete]
+[[painless-count-http]]
+==== Counting HTTP responses by using scripted metric aggregation
+
+You can count the different HTTP response types in a web log data set by using 
+scripted metric aggregation as part of the {transform}. The example below 
+assumes that the HTTP response codes are stored in the `response` field of the 
+documents.
```
Comment (hendrikmuhs) [inline docs/reference/transform/painless-examples.asciidoc]: I would add that the response code is stored as keyword and not as numeric.

**Review by hendrikmuhs — APPROVED** [review]
LGTM

**Comment by szabosteve** [comment]
**Backport commits:**
7.x: a5497cd
7.7: 2d91483

**Pull request merged**